karate-tests/src/test/java/helpers/utils.feature

```diff
@@ -1,45 +1,89 @@
 @ignore
 Feature: Utils
 
-A collection of utility functions...
+A collection of utility functions for generating, validating and manipulating data
 
 Scenario:
-    * def randomUUID = function(){ return java.util.UUID.randomUUID() + '' }
-    * def randomInt = function() { return Math.random() * (10000 - 1) + 1 }
-
-    * def isTodaysDate = 
-    """
-    function(inputDate) {
-      var input = (new Date(Date.parse(inputDate)).toDateString());
-      var today = (new Date()).toDateString();
-      return input == today;
-    }
-    """
-    
-    * def getRandomBirthDate = 
-    """
-    function() {
-      var min = new Date(Date.parse('1930-01-01')).getTime();
-      var max = new Date().getTime();
-      var randomValue = Math.random() * (max - min) + min;
-
-      var randomDate = new Date(randomValue)
-      return randomDate.toISOString().split('T')[0]
-    }
-    """
+  # data generators
+  * def randomUUID = function(){ return java.util.UUID.randomUUID() + '' }
+  * def randomInt = function() { return Math.floor(Math.random() * 999) + 1 }
+  * def randomString = 
+  """
+  function(length) {
+    let result = '';
+    const characters = 'abcdefghijklmnopqrstuvwxyz';
+    const charactersLength = characters.length;
 
-    * def sleep = function(pause){ java.lang.Thread.sleep(pause) }
-    
-    # pick an item at random from an array, ignoring one of the items
-    * def pickDifferentOption = 
-    """
-    function(optionsArray, itemToIgnore) {
-      const index = optionsArray.indexOf(itemToIgnore);
-      optionsArray.splice(index, 1)
-      return optionsArray[Math.floor(Math.random() * optionsArray.length)]
+    for (let i = 0; i < length; i++) {
+        result += characters.charAt(Math.floor(Math.random() * charactersLength));
     }
-    """
+
+    return result;
+  }
+  """
+  * def randomBirthDate = 
+  """
+  function() {
+    var min = new Date(Date.parse('1930-01-01')).getTime();
```

Comment:
should these vars be changed to const or let so consistent with ecma6?

Reply:
Sure, it has no bearing on the functionality but it's an easy change and it doesn't hurt. I've pushed a change